CMakeLists.txt

```diff
@@ -0,0 +1,1042 @@
+		SET(CMAKE_Fortran_LINKER_FLAGS "-m64 ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(64BIT)
+
+	if(NATIVE)
+		SET(CMAKE_Fortran_FLAGS_RELEASE "-march=native -mno-fma4 -mno-fma ${CMAKE_Fortran_FLAGS_RELEASE}")
+		#Nothing to do for DEBUG
+	endif(NATIVE)
+
+	if(STATIC)
+		SET(CMAKE_Fortran_LINKER_FLAGS "-static ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(STATIC)
+
+elseif (Fortran_COMPILER_NAME MATCHES "ifort.*")
+	# ifort
+	set (CMAKE_Fortran_FLAGS_RELEASE "-fp-model source -fp-model strict -fp-model no-except -O3 -no-fma")
+	set (CMAKE_Fortran_FLAGS_DEBUG   "-fp-model source -fp-model strict -fp-model no-except -O0 -g -no-fma")
+
+	if(32BIT)
+		set(CMAKE_Fortran_FLAGS_RELEASE "-m32 ${CMAKE_Fortran_FLAGS_RELEASE}")
+		set(CMAKE_Fortran_FLAGS_DEBUG "-m32 ${CMAKE_Fortran_FLAGS_DEBUG}")
+		SET(CMAKE_Fortran_LINKER_FLAGS "-m32 ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(32BIT)
+
+	if(64BIT)
+		set(CMAKE_Fortran_FLAGS_RELEASE "-m64 ${CMAKE_Fortran_FLAGS_RELEASE}")
+		set(CMAKE_Fortran_FLAGS_DEBUG "-m64 ${CMAKE_Fortran_FLAGS_DEBUG}")
+		SET(CMAKE_Fortran_LINKER_FLAGS "-m64 ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(64BIT)
+
+	if(NATIVE)
+		SET(CMAKE_Fortran_FLAGS_RELEASE "-xHost ${CMAKE_Fortran_FLAGS_RELEASE}")
+		#Nothing to do for DEBUG
+	endif(NATIVE)
+
+	if(STATIC)
+		SET(CMAKE_Fortran_LINKER_FLAGS "-static ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(STATIC)
+
+elseif (Fortran_COMPILER_NAME MATCHES "nagfor.*")
+	# nagfor
+	#Here I am going to assume that since we are using nagfor we are on lxplus for now
+	set (CMAKE_Fortran_FLAGS_RELEASE "-O4 -dusty -ieee=full -dcfuns -w=ulv -w=uparam")
+	set (CMAKE_Fortran_FLAGS_DEBUG   "-O0 -g -dusty -ieee=full -dcfuns -w=ulv -w=uparam")
+
+	if(32BIT)
+		set(CMAKE_Fortran_FLAGS_RELEASE "-abi=32 ${CMAKE_Fortran_FLAGS_RELEASE}")
+		set(CMAKE_Fortran_FLAGS_DEBUG "-abi=32 ${CMAKE_Fortran_FLAGS_DEBUG}")
+		if(STATIC)
+			find_library(g2cLink NAMES libg2c.a PATHS /usr/lib /usr/lib/gcc/x86_64-redhat-linux/3.4.6/32 )
+		else()
+			find_library(g2cLink NAMES libg2c.so PATHS /usr/lib /usr/lib/gcc/x86_64-redhat-linux/3.4.6/32 )
+		endif()
+	endif(32BIT)
+
+	if(64BIT)
+		#See the nagfor man page for the different 64bit -abi options.
+		set(CMAKE_Fortran_FLAGS_RELEASE "-abi=64t ${CMAKE_Fortran_FLAGS_RELEASE}")
+		set(CMAKE_Fortran_FLAGS_DEBUG "-abi=64t ${CMAKE_Fortran_FLAGS_DEBUG}")
+		if(STATIC)
+			find_library(g2cLink NAMES libg2c.a PATHS /usr/lib /usr/lib/gcc/x86_64-redhat-linux/3.4.6 )
+		else()
+			find_library(g2cLink NAMES libg2c.so PATHS /usr/lib /usr/lib/gcc/x86_64-redhat-linux/3.4.6 )
+		endif()
+	endif(64BIT)
+
+	if(NATIVE)
+		SET(CMAKE_Fortran_FLAGS_RELEASE "${CMAKE_Fortran_FLAGS_RELEASE}")
+	endif(NATIVE)
+
+	if(STATIC)
+		SET(CMAKE_Fortran_LINKER_FLAGS "-Bstatic ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(STATIC)
+
+	if(NOT STATIC)
+		SET(CMAKE_Fortran_LINKER_FLAGS "-Bdynamic ${CMAKE_Fortran_LINKER_FLAGS}")
+	endif(NOT STATIC)
+
+#generic catch option
+else (Fortran_COMPILER_NAME MATCHES "gfortran.*")
```

Comment:
```
if()

else()
```
Its to match nested if/else statements. The current versions of cmake don't really need this so it could just be else() with nothing in the brackets.